add docker compose and remove unneeded stuff from template (#1)
iglu-sh/docs: docs/intro.md

---
sidebar_position: 1
---
# Introducing Iglu

## What is Iglu?

> **Iglu** is an easy to use cache for nix derivations. So that you don't have to build your derivations (packages, configurations, ...) multipletimes.

Reviewer: ```suggestion
**Iglu** is an easy to use cache for nix derivations, so that you don't have to build your derivations (packages, configurations, ...) multiple times.
```

Reviewer: Still don't think that this is correct but it's more correct than it was before lmao

You can use the official [cachix](https://hackage.haskell.org/package/cachix) client to push derivations to your own Iglu cache.

## Why Iglu?

> Iglu was build to improve the usability of nix caches for normal users and organisations. We tested much caches, like

Reviewer: ```suggestion
Iglu was build to improve the usability of nix caches for normal users and organisations. We tested some community implementations, like
```

- [attic](https://github.com/zhaofengli/attic/)
- [cachix](https://www.cachix.org/)

> - [cache-server](https://github.com/mifka01/cache-server)

Reviewer: ```suggestion
- [cache-server](https://github.com/mifka01/cache-server)
- [nix-serve](https://github.com/edolstra/nix-serve)
```

> And everytime we get disapointed, either because of the pricing, the usability or bugs.

Reviewer: ```suggestion
Sadly, all of these didn't fit our requirements for a stable, multi-tenant implemenation that could be used with the cachix client.
```

So we ([SirBerg](https://github.com/SirBerg) and [Svenum](https://github.com/Svenum)) decided to build our own cache!
We want to provide these features:

- A simple UI to manage all caches
- A stable and userfriendly experience

> - A way to build and upload derivations easyliy and automated by a builtin [builder](https://github.com/iglu-sh/builder)

Reviewer: ```suggestion
- A way to build and upload derivations easyliy and automated by a builtin [builder](https://github.com/iglu-sh/builder)
- An easy-to-use and easy-to-maintain codebase for either forking or colaboration
```

## Comparison
Here are some comparison to other caches:

|Feature|[Iglu](https://github.com/iglu-sh/controller)|[cachix](https://www.cachix.org/)|[attic](https://github.com/zhaofengli/attic/)|[cache-server](https://github.com/mifka01/cache-server)|
|-------|---------------------------------------------|---------------------------------|---------------------------------------------|-------------------------------------------------------|
|open source            |✅|❌|✅|✅|
|self hostable          |✅|❌|✅|✅|
|multiple caches        |✅|✅|✅|✅|

> |builtin builder        |✅|❌|❌|❌|

Reviewer: Well you **could** use a hydra instance for that...

|cachix compatible      |✅|✅|❌|✅|
|UI for configuration   |✅|✅|❌|❌|
|free of charge         |✅|❌|✅|✅|

> |flexible storag backend|❌|❌|✅|✅|

Reviewer: ```suggestion
|flexible storage backend|❌|❌|✅|✅|
```

Reviewer: I'd like to offer this btw (an S3 compatible storage backend would allow us to easily deploy to Cloudflare R2 for example). And I'd also like to offer a completely serverless variant of the cache, could we create like a "Planned Features" section or something like that?

Author: I add an icon
